bookied/triggers/create.py

```diff
@@ -53,32 +59,48 @@ def createBmgs(self):
             uialist = list()
             uialist = bmg["asset"]
             name = bmg["description"]["en"]
-            x = 0 
+            x = 0
             while x < len(uialist):
-                bmg["asset"] =  uialist[x]
+                bmg["asset"] = uialist[x]
                 if(len(uialist) > 1):
-                   bmg["description"]["en"] = name + "_" + str(bmg["asset"]) 
-                   # To avoid duplicate BMG's on retriggering replays from DP's
-                   if "id" in bmg:
-                      bmg["id"] = None
-                 
+                    bmg["description"]["en"] = name + "_" + str(bmg["asset"])
+                    # To avoid duplicate BMG's on retriggering replays from DP's
+                    if "id" in bmg:
+                        bmg["id"] = None
+
                 x += 1
                 # Skip dynamic bmgs
                 if bmg["dynamic"]:
-                   # Dynamic BMGs are created separately
-                   log.debug("Skipping dynamic BMG: {}".format(str(bmg.identifier)))
-                   continue
+                    # Dynamic BMGs are created separately
+                    log.debug("Skipping dynamic BMG: {}".format(str(bmg.identifier)))
+                    continue
                 bmg.update()
-                self.createBms(bmg)
+                statusCreateBms = self.createBms(bmg)
+                if statusCreateBms == "midway":
+                    print("createBmms, toc > 10")
+                    return 'midway'
+
+            # Terminate loop and return midway if bmg creation took time.
+            # So that incident is processed later.
+            toc = time.time() - tic
+            if toc > 30:
+                return 'midway'
 
     def createBms(self, bmg):
         """ Go through all betting markets and create them
         """
+        tic = time.time()
         for bm in bmg.bettingmarkets:
+
             log.debug(
                 "Updating Betting Market {} ...".format(bm["description"].get("en"))
             )
             bm.update()
+            # break the loop and return midway if bm creation takes time.
+            toc = time.time()
+            if (toc - tic) > 10:
```

Comment:
any criteria for selecting 30  secs over here, levee a comment

Reply:
This is decided based on soccer events. For soccer, the first bmg has six bms. There are cases where sometimes creation of one bm takes about 30 seconds, that an rq time out is reached before the bmg for loop timeout. It's natural to take about 3 seconds for creation of a bm and 30 seconds is not acceptable. Decided somewhere in the middle, 10 seconds.

Reply:
okay, makes sense